### ISSUE TITLE
Prop types error in functional component

### ISSUE THREAD

**Issue by fifn2** [iamturns/eslint-config-airbnb-typescript#57]
I have a file that contains a component
```tsx
const Component: React.FC<ComponentProps> = ({
  reallyLongProp,
  alsoReallyLong
}) => {
  const [count, setCount] = useState(0);
  return (
    <ul>
      <li>{reallyLongProp}</li>
      <li>{alsoReallyLong}</li>
      <li>{count}</li>
      <li>
        <button type="button" onClick={(): void => setCount(count + 1)}>
          Increment
        </button>
      </li>
    </ul>
  );
};

```
And I'm getting these errors:
+ react/prop-types: 'reallyLongProp' is missing in props validation
+ react/prop-types: 'alsoReallyLong' is missing in props validation

**Comment by iamturns** [comment]
The [Airbnb Style Guide](https://airbnb.io/javascript/react/#props) recommends using the `prop-types` package. In the code pasted above, there is no `prop-types` usage. So it seems the linter has done the correct thing here.

If you disagree with this rule from Airbnb, you can disable it within your project. Example ESLint config:

```
module.exports = {
  extends: ["airbnb-typescript"],
  parserOptions: {
    project: "./tsconfig.json"
  },
  rules: {
    "react/prop-types": "off"
  }
}
```

I'm no expert on this rule, and haven't been in React land for a little while. Please let me know if I've missed something here. Cheers!

**Comment by rdsedmundo** [comment]
prop-types has no use case at all if you're using TypeScript—which I'm sure everyone is if they're downloading this package—because you can simply use the static typing capabilities for defining the props a component accepts.

> As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like Flow or TypeScript to typecheck your whole application. But even **if you don’t use those, React has some built-in typechecking abilities.**
https://reactjs.org/docs/typechecking-with-proptypes.html

Although I know the goal of the project is to be as similar as possible to the original Airbnb style guide, IMO this is an exception and should be disabled by default, as it's contradictory/misleading.
